src/scheduler_utils.cc

```diff
@@ -303,38 +316,43 @@ Status
 PriorityQueue::Dequeue(std::unique_ptr<InferenceRequest>* request)
 {
   pending_cursor_.valid_ = false;
-  while (true) {
-    if (!queues_[front_priority_level_].Empty()) {
-      RETURN_IF_ERROR(queues_[front_priority_level_].Dequeue(request));
+  auto it_start = queues_.lower_bound(front_priority_level_);
+  for (auto it = it_start; it != queues_.end(); ++it) {
```

Comment:
Shorter for loop for consistency.

Reply:
A range-based for loop (e.g. for (auto& queue : queues_)) will range from begin to end and not benefit from the (small) optimization using it_start (found in logarithmic time using front_priority_level_).

A suggestion for a shorter for loop:
auto it = queues_.lower_bound(front_priority_level_);
for (; it != queues_.end(); ++it) {
  ...
}

Reply:
Or we can drop the use of "front_priority_level_", as the request to be dequeued is likely to be found in the first key values of "queues_" (excepted when there is a large amount of user-specified non-default PolicyQueue)

Reply:
Thanks for explaining and addressing the other loop. This looks good to me.

I'm less familiar with these files, so I'll wait for @GuanLuo to give this PR a once-over as well,

Reply:
Let's keep `front_priority_level_` for now to minimize change until necessary, but I think `last_priority_level_` can be safely removed as it is no longer used anywhere.